tests/ui/const_is_empty.rs

```diff
@@ -0,0 +1,156 @@
+#![warn(clippy::const_is_empty)]
+#![allow(clippy::needless_late_init, unused_must_use)]
+
+fn test_literal() {
```

Comment:
With the new constant evaluation, could you also add two tests, like this:

```
#![feature(inline_const)]

const fn const_rand() -> &'static str {
    "17"
}

fn const_expressions() {
    let _ = const { if true { "1" } else { "2" }}.is_empty();

    let _ = const_rand().is_empty();
}
```

Reply:
Done, but note that they will not trigger the lint. The full constant propagation hasn't happened yet from what I've understood, which is why we try to follow simple constant declarations through local variables and const variables declarations.